tests/test_sim_trait.py

```diff
@@ -0,0 +1,775 @@
+import functools
+
+import msprime
+import numpy as np
+import pytest
+import tskit
+import tstrait
+
+
+@functools.lru_cache(maxsize=None)
+def all_trees_ts(n):
+    """
+    Generate a tree sequence that corresponds to the lexicographic listing
+    of all trees with n leaves (i.e. from tskit.all_trees(n)).
+
+    Note: it would be nice to include a version of this in the combinatorics
+    module at some point but the implementation is quite inefficient. Also
+    it's not entirely clear that the way we're allocating node times is
+    guaranteed to work.
+    """
+    tables = tskit.TableCollection(0)
+    for _ in range(n):
+        tables.nodes.add_row(flags=tskit.NODE_IS_SAMPLE, time=0)
+    for j in range(1, n):
+        tables.nodes.add_row(flags=0, time=j)
+
+    L = 0
+    for tree in tskit.all_trees(n):
+        for u in tree.preorder()[1:]:
+            tables.edges.add_row(L, L + 1, tree.parent(u), u)
+        L += 1
+    tables.sequence_length = L
+    tables.sort()
+    tables.simplify()
+    return tables.tree_sequence()
+
+
+class Test_sim_phenotype_output_dim:
+    @pytest.mark.parametrize("num_ind", [1, 2, np.array([5])[0]])
+    @pytest.mark.parametrize("num_causal", [1, 2, np.array([3])[0]])
+    @pytest.mark.parametrize("alpha", [0, 1, -1.1])
+    @pytest.mark.parametrize("random_seed", [1, 2, None])
+    def test_output_dim_normal(self, num_ind, num_causal, alpha, random_seed):
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        ts = msprime.sim_ancestry(
+            num_ind, sequence_length=100_000, random_seed=random_seed
+        )
+        ts = msprime.sim_mutations(ts, rate=0.01, random_seed=random_seed)
+        sim_result = tstrait.sim_traits(
+            ts=ts,
+            num_causal=num_causal,
+            model=model,
+            alpha=alpha,
+            random_seed=random_seed,
+        )
+        assert sim_result.shape[0] == num_causal
+        assert sim_result.shape[1] == 3
+        assert len(sim_result["SiteID"]) == num_causal
+        assert len(sim_result["CausalState"]) == num_causal
+        assert len(sim_result["EffectSize"]) == num_causal
+
+    def test_binary_mutation_model(self):
+        random_seed = 1
+        num_ind = 10
+        num_causal = 5
+        alpha = 1
+        model = tstrait.trait_model(distribution="normal", mean=0, var=1)
+        ts = msprime.sim_ancestry(
+            num_ind, sequence_length=100_000, random_seed=random_seed
+        )
+        ts = msprime.sim_mutations(
+            ts, rate=0.01, random_seed=random_seed, model="binary"
+        )
+        sim_result = tstrait.sim_traits(
+            ts=ts,
+            num_causal=num_causal,
+            model=model,
+            alpha=alpha,
+            random_seed=random_seed,
+        )
+        assert sim_result.shape[0] == num_causal
```

Comment:
These tests aren't very strong, and there's a lot of duplicated code. You could delegate most of this to a method 
```
def check_dimensions(self, df, num_causal):
    assert len(df) == num_causal  
    # etc